clients/drcachesim/scheduler/scheduler.h

```diff
@@ -829,6 +829,16 @@ template <typename RecordType, typename ReaderType> class scheduler_tmpl_t {
          * when raising this value on uneven inputs.
          */
         double exit_if_fraction_inputs_left = 0.1;
+        /**
+         * Adds noise generators to the scheduler. A noise generator creates synthetic
+         * trace records that will be scheduled alongside records of one or more real
+         * traces. Each noise generator behaves like a thread in its own process.
+         */
+        uint64_t noise_generator_add = 0;
+        /**
+         * Number of synthetic trace records produced by the noise generator.
```

Comment:
This is harder to change than the runtime option: once this field is here we can't remove it.  While incrementally adding pieces is a good approach, the parts that form compatibility-sensitive interfaces need more care as we can't just change these incrementally without consequences.

Reply:
Thinking about this more: we want the synthetic workloads to be able to be given lower priorities (simulating batch workloads), pinned to certain cores, etc.   I think for the scheduler interface the user would create an input_workload_t for each synthetic workload, with a vector of readers (there's the control over threads per process) and desired scheduling flags like priority.

So I don't think we want fields like these here in the global options: instead we'd want a way to create a generator as a reader_t so we can wrap an input_workload_t around it.

Reply:
As discussed offline, we now have only one global option to enable the noise generator.
Other parameters to drive the noise generator are now in `noise_generator_info_t`.
For now they just have some default value, the plan is to set those analyzing the trace scheduled alongside the noise.

The noise generators are also added at the analyzer level, rather than at the scheduler level as discussed.
So the noise generator is somewhat transparent to the scheduler, and it's just another workload.
We can do the same internally.

Reply:
Outdated.
Using `noise_generator_factory_t` to add `noise_generator_t` to scheduler's input workloads.